services/src/main/java/org/keycloak/exportimport/util/ExportUtils.java

```diff
@@ -424,7 +424,7 @@ private static PolicyRepresentation createPolicyRepresentation(AuthorizationProv
      * @return RoleRepresentation with all stuff filled (including composite roles)
      */
     public static RoleRepresentation exportRole(RoleModel role) {
-        RoleRepresentation roleRep = ModelToRepresentation.toRepresentation(role);
+        RoleRepresentation roleRep = ModelToRepresentation.toBriefRepresentation(role);
```

Comment:
I believe export should contain full role representation. This also impacts the export/import tests that need to be added, see org.keycloak.testsuite.exportimport.ExportImportTest.

Reply:
So far I renamed the existing `toRepresentation` to `toBriefRepresentation` except for the GET call for a role resource. I could have a look into the `exportRole` function you mentioned as it sounds like you're right. It would make sense to export the full detail role representation. Before I update the commit are there any other places that would need to have the full `toRepresentation` call?

Reply:
I checked the rest and it looks good to me. I did not realize that one earlier so it was really good that you performed the renaming that revealed this issue. The full `toRepresentation` should be used for the export/import and role detail only, not for any other place. So IMHO, the rest of the code should stay as it is.

Reply:
I was looking for an import function mapping representations to models. So far the closest thing I found was https://github.com/keycloak/keycloak/blob/master/server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java#L1033 but that looks to me like it only imports the name and id... Is that right? Also is there another place? The previously discussed `exportRole` I'll fix.